### DISCUSSION TITLE
BC for Use in a Postprocessor

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#28721] [Q&A Modules: General] [answered]
How do I resolve this error:
*** ERROR ***
Task add_postprocessor is not registered to build BoundaryCondition derived objects

**Comment by GiudGiud** [comment]
You move the boundary condition you placed in the `[Postprocessors]` block into the `[BCs]` block

**Reply by TLWise** [reply]
Thank you. The example code on the website (https://mooseframework.inl.gov/syntax/Postprocessors/index.html#12778715-26f2-4c0c-b586-10bd52d0368e) may not reflect the proper means to accomplish using this postprocessor.
The previous error went away, but was replaced with this one: 
*** ERROR ***
unused parameter 'BCs/totalFlux/coupled_scalar'

**Reply by GiudGiud** [reply]
which object are you using with this parameter?

**Reply by GiudGiud** [reply]
ScalarCoupledPostprocessor is a test object, you generally cant use it without adding --allow-test-objects on the command line

**Reply by TLWise** [reply]
IntegratedBC

**Reply by GiudGiud** [reply]
IntegratedBC is a base class you could not be using this in an input file.
Could you please paste your BCs block here?

**Reply by TLWise** [reply]
```
[Mesh]
  type = GeneratedMesh
  dim = 2
  nx = 5
  ny = 5
  xmax = 1
  ymax = 1
  elem_type = QUAD4
[]

[Variables]
  [./u]
    initial_condition = 1
  [../]
  [./scalar_variable]
    family = SCALAR
    order = FIRST
    initial_condition = 2
  [../]
[]

[Kernels]
  [./diff]
    type = Diffusion
    variable = u
  [../]
[]

[ScalarKernels]
  [./td1]
    type = ODETimeDerivative
    variable = scalar_variable
  [../]
[]

[BCs]
  [./leftDirichlet]
      type = DirichletBC
      variable = u
      boundary = 'left'
      value = 1
  [../]
  [./rightDirichlet]
      type = DirichletBC
      variable = u
      boundary = 'right'
      value = 0
  [../]
[]

[Postprocessors]
  [./totalFlux]
    type = ScalarCoupledPostprocessor
    variable = u
    coupled_scalar = scalar_variable
    boundary = left
  [../]
[]

[Executioner]
  type = Transient
  dt = 1
  num_steps = 1
  solve_type = JFNK
  l_max_its = 30
  l_tol = 1e-6
  nl_max_its = 20
  nl_rel_tol = 1e-5
[]

[Outputs]
  csv = true
[]
```

**Reply by GiudGiud** [reply, marked as answer]
the issue is here
```
[Postprocessors]
  [./totalFlux]
    type = ScalarCoupledPostprocessor
    variable = u
    coupled_scalar = scalar_variable
    boundary = left
  [../]
[]
```
this is a test object. unless you are using the moose executable with --allow-test-objects it wont work
use another postprocessor to couple in a scalar variable

**Reply by TLWise** [reply]
Got it. Thank you.
